numpy/linalg/umath_linalg.c.src

```diff
@@ -2103,13 +2111,18 @@ process_@lapack_func@_results(GEEV_PARAMS_t *params)
 
 /**end repeat**/
 
+/*
+ * TODO:
+ * reinstate the CFLOAT, COMPLEX_t, fortran_complex, float, cggev variant
+ * for single floats and use it in eig, eigval
+ */
 
 /**begin repeat
-   #TYPE = CFLOAT, CDOUBLE#
-   #typ = COMPLEX_t, DOUBLECOMPLEX_t#
-   #ftyp = fortran_complex, fortran_doublecomplex#
-   #realtyp = float, double#
-   #lapack_func = cgeev, zgeev#
+   #TYPE = CDOUBLE#
+   #typ = DOUBLECOMPLEX_t#
+   #ftyp = fortran_doublecomplex#
+   #realtyp = double#
+   #lapack_func = zgeev#
```

Comment:
It turns out the `cfloat` loops were not used. The python function upcasts to `cdouble`. I am not sure why they were added in on commit in gh-3220, but then neutralized in another.

Reply:
Oh, interesting. Looking yet another time, the python code specifically uses double precision always, but then casts back to single precision. Is there something specific about eig that makes single precision numerically bad? Otherwise it seems like maybe we should register all loops and remove forcing the signature to double.

Reply:
From what I've seen, the other gufuncs register all the loops. and the deliberately avoid using them in python. Can we do the same thing here?

Reply:
removed this from the PR, will reintroduce a cleaner solution in a future one